src/matchRoute.js

```diff
@@ -0,0 +1,69 @@
+/**
+ *
+ * @param {import('./constructRoutes').ResolvedRoutesConfig} resolvedRoutesConfig
+ * @param {string} path
+ * @returns {import('./constructRoutes').ResolvedRoutesConfig}
+ */
+export function matchRoute(resolvedRoutesConfig, pathMatch) {
+  const result = { ...resolvedRoutesConfig };
+
+  const baseWithoutSlash = resolvedRoutesConfig.base.slice(
+    0,
+    resolvedRoutesConfig.base.length - 1
+  );
+
+  if (pathMatch.startsWith(baseWithoutSlash)) {
+    result.routes = recurseRoutes(
+      pathMatch,
+      resolvedRoutesConfig.base,
+      resolvedRoutesConfig.routes
+    );
+  } else {
+    result.routes = [];
+  }
+
+  return result;
+}
+
+/**
+ *
+ * @param {string} pathMatch
+ * @param {string} startPath
+ * @param {Array<import('./constructRoutes').Route>} routes
+ */
+function recurseRoutes(pathMatch, startPath, routes) {
```

Comment:
I would personally appreciate seeing a test for this method directly. 

I see it's run when we test `matchRoute` so it's probably not necessary and feel free to ignore this feedback.

Reply:
I prefer testing through the public API - that way we can change implementation details while verifying that we haven't made a breaking change to the public API. The `recurseRoutes` function is heavily tested through the matchRoutes function, so I think I'll stick with those tests.